tests/test_chatmember.py

```diff
@@ -17,50 +17,92 @@
 # You should have received a copy of the GNU Lesser Public License
 # along with this program.  If not, see [http://www.gnu.org/licenses/].
 import datetime
+from copy import deepcopy
 
 import pytest
 
-from telegram import User, ChatMember
 from telegram.utils.helpers import to_timestamp
+from telegram import (
+    User,
+    ChatMember,
+    ChatMemberOwner,
+    ChatMemberAdministrator,
+    ChatMemberMember,
+    ChatMemberRestricted,
+    ChatMemberLeft,
+    ChatMemberBanned,
+    Dice,
+)
 
 
 @pytest.fixture(scope='class')
 def user():
     return User(1, 'First name', False)
 
 
+@pytest.fixture(
+    scope="class",
+    params=[
+        (ChatMemberOwner, ChatMember.CREATOR),
+        (ChatMemberAdministrator, ChatMember.ADMINISTRATOR),
+        (ChatMemberMember, ChatMember.MEMBER),
+        (ChatMemberRestricted, ChatMember.RESTRICTED),
+        (ChatMemberLeft, ChatMember.LEFT),
+        (ChatMemberBanned, ChatMember.KICKED),
+    ],
+    ids=[
+        ChatMember.CREATOR,
+        ChatMember.ADMINISTRATOR,
+        ChatMember.MEMBER,
+        ChatMember.RESTRICTED,
+        ChatMember.LEFT,
+        ChatMember.KICKED,
+    ],
+)
+def chatmember_class_and_status(request):
+    return request.param
+
+
 @pytest.fixture(scope='class')
-def chat_member(user):
-    return ChatMember(user, TestChatMember.status)
+def chatmember_types(chatmember_class_and_status, user):
+    return chatmember_class_and_status[0](status=chatmember_class_and_status[1], user=user)
 
 
 class TestChatMember:
-    status = ChatMember.CREATOR
-
-    def test_slot_behaviour(self, chat_member, recwarn, mro_slots):
-        for attr in chat_member.__slots__:
-            assert getattr(chat_member, attr, 'err') != 'err', f"got extra slot '{attr}'"
-        assert not chat_member.__dict__, f"got missing slot(s): {chat_member.__dict__}"
-        assert len(mro_slots(chat_member)) == len(set(mro_slots(chat_member))), "duplicate slot"
-        chat_member.custom, chat_member.status = 'should give warning', self.status
+    def test_slot_behaviour(self, chatmember_types, mro_slots, recwarn):
+        for attr in chatmember_types.__slots__:
+            assert getattr(chatmember_types, attr, 'err') != 'err', f"got extra slot '{attr}'"
+        assert len(mro_slots(chatmember_types)) == len(
+            set(mro_slots(chatmember_types))
+        ), "duplicate slot"
+        chatmember_types.custom, chatmember_types.status = 'warning!', chatmember_types.status
         assert len(recwarn) == 1 and 'custom' in str(recwarn[0].message), recwarn.list
+        with pytest.raises(AttributeError):
+            chatmember_types.custom2
 
-    def test_de_json_required_args(self, bot, user):
-        json_dict = {'user': user.to_dict(), 'status': self.status}
+    def test_de_json_required_args(self, bot, chatmember_class_and_status, user):
```

Comment:
note to self: add a test making sure that `ChatMemberSubclass.de_json` only ever returns a `ChatMemberSubclass` object even when passed kwargs that suit a different subclass. same for botcommandscope